src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -272,14 +280,40 @@ export class DataSourceSavedObjectsClientWrapper {
             '"auth.credentials.password" attribute is required'
           );
         }
+        break;
+      case AuthType.SigV4:
+        if (!credentials) {
+          throw SavedObjectsErrorHelpers.createBadRequestError(
+            '"auth.credentials" attribute is required'
+          );
+        }
+
+        const { accessKey, secretKey, region } = credentials as SigV4Content;
+
+        if (!accessKey) {
+          throw SavedObjectsErrorHelpers.createBadRequestError(
+            '"auth.credentials.accessKey" attribute is required'
+          );
+        }
+
+        if (!secretKey) {
+          throw SavedObjectsErrorHelpers.createBadRequestError(
+            '"auth.credentials.secretKey" attribute is required'
+          );
+        }
 
+        if (!region) {
+          throw SavedObjectsErrorHelpers.createBadRequestError(
+            '"auth.credentials.region" attribute is required'
+          );
+        }
         break;
       default:
         throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${type}'`);
     }
   }
 
-  private async validateEncryptionContext(
+  private async getEncryptionContext(
```

Comment:
in case of getting encryption context, I think we don't need to do following validation, right?

Reply:
rename and restructure this part to make code readable